meilisearch/tests/documents/add_documents.rs

```diff
@@ -896,16 +896,104 @@ async fn error_primary_key_inference() {
     index.wait_task(0).await;
     let (response, code) = index.get_task(0).await;
     assert_eq!(code, 200);
-    assert_eq!(response["status"], "failed");
 
-    let expected_error = json!({
-        "message": r#"The primary key inference process failed because the engine did not find any fields containing `id` substring in their name. If your document identifier does not contain any `id` substring, you can set the primary key of the index."#,
-        "code": "primary_key_inference_failed",
+    insta::assert_json_snapshot!(response, { ".duration" => "[duration]", ".enqueuedAt" => "[date]", ".startedAt" => "[date]", ".finishedAt" => "[date]" },
+    @r###"
+    {
+      "uid": 0,
+      "indexUid": "test",
+      "status": "failed",
+      "type": "documentAdditionOrUpdate",
+      "canceledBy": null,
+      "details": {
+        "receivedDocuments": 1,
+        "indexedDocuments": 1
+      },
+      "error": {
+        "message": "The primary key inference process failed because the engine did not find any field ending with `id` in its name. Please specify the primary key manually using the `primaryKey` query parameter.",
```

Comment:
@dichotommy, can you review this message that occurs when inference fails due to no candidate field?

Reply:
the changes will be handled in https://github.com/meilisearch/meilisearch/issues/3301

Reply:
Maybe ready for rc0, otherwise will be present in rc1 💪